blaseball_mike/chronicler.py

```diff
@@ -21,26 +21,42 @@ def prepare_id(id_):
         raise ValueError(f'Incorrect ID type: {type(id_)}')
 
 
-def paged_get(url, params, session):
+def paged_get(url, params, session,lazy=False):
     """
     Combine paged URL responses
     """
+    if lazy:
+        return paged_get_lazy(url,params,session)
+
     data = []
     while True:
         out = check_network_response(session.get(url, params=params))
         d = out.get("data", [])
         page = out.get("nextPage")
-        
+
         data.extend(d)
         if page is None or len(d) == 0 or params.get("count", 1000) >= len(d):
             break
         params["page"] = page
 
     return data
 
+def paged_get_lazy(url,params,session):
+    """
+    Combine paged URL responses; returns a generator
+    """
+    while True:
+        out = check_network_response(session.get(url, params=params))
+        d = out.get("data", [])
+        page = out.get("nextPage")
+
+        yield from d
+        if page is None or len(d) == 0 or params.get("count", 1000) >= len(d):
```

Comment:
i think you might lose the last page here

Reply:
i don't think it will?
```py
        data.extend(d) | yield from d -> gets all results from curr. page
        if page is None or len(d) == 0 or params.get("count", 1000) >= len(d):
            break | return -> breaks loop; all data has been added/returned
        params["page"] = page

    return data
```
might be missing something obvious though

Reply:
i might also be off by one